### PR TITLE
extract jq debug output

### PR DESCRIPTION
**Pull request by glennj**

### CODE REVIEW

File: tests/data/with_debug/bats-jq.bash
Comment (glennj) [inline tests/data/with_debug/bats-jq.bash:1]: This is the file that will be dropped into every jq exercise dir.

File: tests/tests.bats [hunk 1]
```diff
@@ -89,6 +89,14 @@ actual_matches_expected() {
     [[ "$status" -eq 1 ]]
 }
 
+@test "with_debug test" {
+    TEST="with_debug"
+    TEST_DIR="$DATA_DIR/$TEST"
+    run "$RUN_SCRIPT" "$TEST" "$TEST_DIR" "$TEST_DIR"
+
+    [[ "$status" -eq 0 ]]
```
Comment (IsaacG) [inline tests/tests.bats:97]: `((` ought to be used for numeric tests.

**Review by IsaacG — APPROVED** [review]

**Comment by glennj** [comment]
I found a problem: it seems to swallow any stderr from jq (i.e. halt_error messages). Working on it.

File: tests/tests.bats (hunk 1 above)
Comment (glennj) [reply tests/tests.bats:97]: true, but "wontfix"

File: tests/data/with_debug/bats-jq.bash [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    while IFS= read -r line || [[ -n $line ]]; do
+        if [[ $line == '["DEBUG:",'* ]]; then
+            echo "# $line" >&3
+        else
+            echo "$line" >&2
+        fi
+    done < "$stderr"
+    rm "$stderr"
+    echo "$output"
+    return $rc
```
Comment (IsaacG) [inline tests/data/with_debug/bats-jq.bash:28]: Could you quote all expansions?

**Review by IsaacG — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**